tests/vmi_configuration_test.go

```diff
@@ -2294,18 +2294,34 @@ var _ = Describe("Configurations", func() {
 	Context("[rfe_id:2926][crit:medium][vendor:cnv-qe@redhat.com][level:component]Check SMBios with default and custom values", func() {
 
 		var vmi *v1.VirtualMachineInstance
+		var originalSmbios string
+
+		cfgMap, err := tests.GetKubeVirtConfigMap()
+		Expect(err).ToNot(HaveOccurred())
+		if val, ok := cfgMap.Data[virtconfig.SmbiosConfigKey]; ok {
+			originalSmbios = val
+		}
 
 		BeforeEach(func() {
 			vmi = tests.NewRandomFedoraVMIWithDmidecode()
 		})
 
+		AfterEach(func() {
+			tests.UpdateClusterConfigValueAndWait(virtconfig.SmbiosConfigKey, originalSmbios)
```

Comment:
Getting closer!

I think we should move this save/restore to the top-level describe for the whole module. a lot of tests are currently violating this restoration principle--we can fix all of them at once.

tests/operator_test.go has a really elegant trick worth incorporating here. That code tests like this:

`curKubeVirtConfig.ResourceVersion != originalKubeVirtConfig.ResourceVersion`

and if the configs don't match, it simply restores the entire configMap data. Any change anywhere in the configMap will be detected and corrected by this. Check the AfterEach on or around line 556.

Doing the above sidesteps another subtle issue: what about clusters where the smbios code wasn't present? Setting `SmbiosConfigKey` like this line does means that if the key wasn't set prior to the test, it will transition from not present at all to the empty string when this test is run. resetting the entire `data` stanza doesn't have that problem.

Reply:
@stu-gott @ashleyschuett we need similar reset of value for machinetype as well

Reply:
@stu-gott @ashleyschuett we need similar reset of value for machinetype as well

Reply:
In this test or?

Reply:
https://github.com/kubevirt/kubevirt/blob/2b35c0cf0f7657b9ad8b1c7a50ed642ca31bfc20/tests/vmi_configuration_test.go#L1538

Here also we need a similar mechanism to restore values to previous state

Reply:
@vatsalparekh is the default machine type fix for *this* test or a related one in this module?